### DISCUSSION TITLE
Image upload is working but I can't retrieve image when editing form

### DISCUSSION THREAD

**Discussion by grafxflow** [livewire/livewire#4126] [1. Help] [answered]
I have created a form that allows an image to be uploaded to google cloud and it works fine, even when going back to edit the form it retrieves the image.

But now I have placed the form in a subdomain url for a CMS and now it only allows the uploading of the image, but it won't load it back when editing the form.

https://website.com (works for uploading and editing)
https://cms.website.com (works for uploading but not for editing)

Is there a security issue that blocks such things?

It is going through cloud flair to a google computer engine.

**Comment by joshhanley** [comment]
@grafxflow in theory if you have a storage driver setup for your google cloud then it should work ok. Maybe it could be a CORS issue in your app possibly?

**Reply by grafxflow** [reply]
I am getting the following output in the console.

```
Origin http://cms.website.test is not allowed by Access-Control-Allow-Origin.

[Error] Fetch API cannot load http://website.test/storage/media/1094/conversions/WECMs6SOUu8avqKJG76Wp3jo4F1Zlc-metaMTg1X2dfOTAwMV8xNl8xLmpwZw==--article.jpg due to access control checks.

Failed to load resource: Origin http://cms.website.test is not allowed by Access-Control-Allow-Origin.
```

The only way I have found to get this to work is changing the **.env** but the problem is that it uses wildcard subdomains from the database.

```
APP_URL=http://cms.website.test
```

**Reply by joshhanley** [reply]
@grafxflow yep definitely a CORS issue, so you will need to change your CORS config to allow both `cms.website.test` and `website.test`.

How to do that though, I'm not exactly sure. I think every setup is different. Such as if you use S3 you would have to set it in S3 for example. Just have a search around for setting CORS in your CMS or Laravel or whatever you are using.

Hope this helps!

**Comment by grafxflow** [comment, marked as answer]
@joshhanley - managed to solve the problem by editing the following file.

**app/Http/Middleware/TrustProxies.php**

```
protected $proxies = '*';
```

**Reply by joshhanley** [reply]
@grafxflow ah nice! Thanks for sharing 🙂
